client/src/portal/mod.rs

```diff
@@ -679,6 +710,82 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn load() -> Result<(), Error> {
+        let temp_dir = tempdir()?;
+        let keyring_dir = temp_dir.path().join("keyrings");
+
+        fs::create_dir_all(&keyring_dir).await?;
+
+        let fixture_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+            .join("fixtures")
+            .join("latest.keyring");
+
+        fs::copy(&fixture_path, &keyring_dir.join("latest.keyring")).await?;
+
+        std::env::set_var("XDG_DATA_HOME", &temp_dir.path());
+
+        // Keyring::load with the wrong secret
+        let password = b"wrong";
+        let secret = Secret::from(password.to_vec());
+        match Keyring::load(keyring_dir.join("latest.keyring"), secret).await {
+            Err(err) => {
+                assert!(matches!(err, Error::ChecksumMismatch));
+            }
+            Ok(_) => assert!(false),
+        };
+
+        // Keyring::load with the correct secret
+        let password = b"test";
+        let secret = Secret::from(password.to_vec());
+        match Keyring::load(keyring_dir.join("latest.keyring"), secret).await {
+            Err(err) => {
+                assert!(matches!(err, Error::ChecksumMismatch));
+            }
+            Ok(_) => assert!(true),
+        };
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn repeated_open() -> Result<(), Error> {
```

Comment:
where this test came from?

Reply:
I am not sure in general what all those tests are intended to verify. You should add one test here in general, which tries to open a fixture file that exists already with the wrong password which would be passing when it shouldn't. The next commit should do the changes for the test's behaviour to be fixed.

Reply:
this test comes from a separate commit, after introducing `verify_secret()`. apologies if this isn't clearer.

Reply:
> why we need it

repeated calls to `oo7::portal::Keyring::open()` failing to identify/validate incorrect keyring secret prompted this entire pr.

so, oo7_daemon uses `Keyring::open()` during `oo7_daemon -l` command (-l option reads keyring secret from stdin).

currently during the very first call to `Keyring::open()` everything works as expected (keyring migration validates the keyring secret). after that, calls to `Keyring::open()` succeed (so `oo7_daemon -l` as well) even with an incorrect keyring secret. this is what I'm trying to fix and this test make sure that introducing `verify_secret()` fixes that issue.